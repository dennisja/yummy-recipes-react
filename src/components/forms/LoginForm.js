import React, { Component } from 'react';
import PropTypes from 'prop-types';
import axios from 'axios';
import Configs from "../../configs/Configs"
class LoginForm extends Component {
  static initialState = {
    email: '',
    password: ''
  }

  constructor(props) {
    super(props);
    this.state = LoginForm.initialState;
  }

  handleInputChange = (event) => {
    const target = event.target;
    this.setState({
      [target.name]: target.value,
    })
  }

  handleLoginSubmit = (event) => {
    event.preventDefault();
    const { onLoginSubmit } = this.props;
    const { baseUrl, loginUrl } = Configs.api
    const form = event.target;
    //submit form
    axios({
      method: "POST",
      url: `${baseUrl}${loginUrl}`,
      auth: {
        username: this.state.email,
        password: this.state.password
      }
    })
      .then(response => {
        onLoginSubmit(response.data);
        this.setState(LoginForm.initialState);
        form.reset();
      })
      .catch(error => {
        const { data, status } = error.response;
        //data has the errors arra
        if ("errors" in data) {
          alert(data.errors);
        }
        console.log(data);
        console.log(status);
      })
  }

  render() {

    return (
      <form onSubmit={this.handleLoginSubmit}>
        <div className="input-field">
          <label htmlFor="email">Email </label>
          <input type="email" name="email" id="email" onChange={this.handleInputChange} required/>
        </div>
        <div className="input-field">
          <label htmlFor="password">Password</label>
          <input type="password" name="password" id="password" onChange={this.handleInputChange} required/>
        </div>
        <div className="center-align">
          <input type="submit" className="btn" name="login" id="login" value="Login" />
        </div>
      </form>
    );
  }
}

LoginForm.propTypes = {
  onLoginSubmit: PropTypes.func.isRequired,
};

export default LoginForm;
