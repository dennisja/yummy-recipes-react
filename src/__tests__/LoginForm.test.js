import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import LoginForm from '../components/forms/LoginForm';

describe('Test RegisterForm', () => {
    const formWrapper = shallow(<LoginForm onLoginSubmit={jest.fn()} />);
    const preventDefault = jest.fn();


    it('renders properly', () => {
        expect(shallowToJson(formWrapper)).toMatchSnapshot();
    });

    it('renders a form element', () => {
        expect(formWrapper.find('form')).toHaveLength(1);
    });

    it('renders the correct form fields', () => {
        expect(formWrapper.find('#email').length).toBe(1);
        expect(formWrapper.find('#password').length).toBe(1);
        expect(formWrapper.find('#login').length).toBe(1);
    });

    it('submits data', () => {
        formWrapper.find('form').simulate('submit', { preventDefault });
        expect(preventDefault).toBeCalled();
    });
});
