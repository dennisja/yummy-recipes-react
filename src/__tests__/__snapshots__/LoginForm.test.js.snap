// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test LoginForm renders properly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="input-field"
  >
    <label
      htmlFor="email"
    >
      Email
    </label>
    <input
      id="email"
      name="email"
      onChange={[Function]}
      required={true}
      type="email"
      value=""
    />
  </div>
  <div
    className="input-field"
  >
    <label
      htmlFor="password"
    >
      Password
    </label>
    <input
      id="password"
      name="password"
      onChange={[Function]}
      required={true}
      type="password"
      value=""
    />
  </div>
  <div
    className="center-align"
  >
    <input
      className="btn btn-small orange"
      id="login"
      name="login"
      type="submit"
      value="Login"
    />
  </div>
</form>
`;
